packages/fuels-abigen-macro/tests/test_projects/nested_structs/src/main.sw

```diff
@@ -8,9 +10,23 @@ pub struct AllStruct {
     some_struct: SomeStruct,
 }
 
+pub struct CallData {
+    memory_address: MemoryAddress,
+    num_coins_to_forward: u64,
+    asset_id_of_coins_to_forward: ContractId,
+    amount_of_gas_to_forward: u64,
+}
+
+struct MemoryAddress {
+    contract_id: ContractId,
```

Comment:
How would is_sway_native_type handle an Identity here?  Is the ABI able to understand that is it either an Address or a ContractId?

Reply:
No, IIRC `Identity` was recently added, not sure how it's behaving now. Do you mind testing it out? If it fails similarly as it did with `Address` before, I can quickly add it to the list of reserved keyword 😄

Reply:
```
contract;

use std::identity::Identity;

pub struct CallData {
    memory_address: MemoryIdentity,
    num_coins_to_forward: u64,
    asset_id_of_coins_to_forward: Identity,
    amount_of_gas_to_forward: u64,
}

struct MemoryIdentity {
    contract_id: Identity,
    function_selector: u64,
    function_data: u64,
}

abi NestedStructs {
    fn dummy_func(call_data: CallData) -> bool;
    fn nested_struct_with_reserved_keyword_substring(call_data: CallData) -> CallData;
}

impl NestedStructs for Contract {
    fn dummy_func(call_data: CallData) -> bool {
        true
    }
    fn nested_struct_with_reserved_keyword_substring(call_data: CallData) -> CallData {
        call_data
    }
}
```
and
```
use fuels::{prelude::*, tx::ContractId};
use fuels_abigen_macro::abigen;

// Load abi from json
abigen!(NestedStructs, "out/debug/nested_structs-abi.json");

async fn get_contract_instance() -> (NestedStructs, ContractId) {
    // Launch a local network and deploy the contract
    let wallet = launch_provider_and_get_single_wallet().await;

    let id = Contract::deploy("./out/debug/nested_structs.bin", &wallet, TxParameters::default())
        .await
        .unwrap();

    let instance = NestedStructs::new(id.to_string(), wallet);

    (instance, id)
}

#[tokio::test]
async fn can_get_contract_id() {
    let (_instance, _id) = get_contract_instance().await;

    // Now you have an instance of your contract you can use to test each function
    let mem_address = nestedstructs_mod::MemoryIdentity {
        contract_id: nestedstructs_mod::Identity::ContractId(ContractId::zeroed()),
        function_selector: 0,
        function_data: 0,
    };

    let call_data = nestedstructs_mod::CallData {
        memory_address: mem_address,
        num_coins_to_forward: 0,
        asset_id_of_coins_to_forward: nestedstructs_mod::Identity::ContractId(_id),
        amount_of_gas_to_forward: 20000,
    };

    let actual = _instance
        .nested_struct_with_reserved_keyword_substring(call_data.clone())
        .call()
        .await
        .unwrap()
        .value;

    assert_eq!(actual, call_data);
}
```
Work!